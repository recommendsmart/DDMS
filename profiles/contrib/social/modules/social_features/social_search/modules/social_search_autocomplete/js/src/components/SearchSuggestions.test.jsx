import React from "react";
import renderer from "react-test-renderer";
import SearchSuggestions from "./SearchSuggestions";

describe("SearchSuggestions", () => {
  const suggestions = [
    {
      type: "Topic",
      label: "Open Social is great for customisable communities",
      url: "http://example.com/node/1",
    },
    {
      type: "Event",
      label: "Open Social launching party",
      url: "http://example.com/node/2",
      tags: ["You are enrolled to this event"],
    },
    {
      type: "Event",
      label: "GoalGorilla anniversary",
      url: "http://example.com/node/3",
    },
  ];

  it("renders nothing when no query is provided", () => {
    const tree = renderer.create(
      <SearchSuggestions
        searchBase="http://example.com/search/base"
        suggestions={suggestions}
      />
    );

    expect(tree.toJSON()).toBeNull();
  });

  it("renders nothing when no suggestions are provided", () => {
    const tree = renderer.create(
      <SearchSuggestions
        searchBase="http://example.com/search/base"
        query="social"
      />
    );

    expect(tree.toJSON()).toBeNull();
  });

  it("renders a list of suggestions", () => {
    const tree = renderer
      .create(
        <SearchSuggestions
          searchBase="http://example.com/search/base"
          query="open"
          suggestions={suggestions}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
