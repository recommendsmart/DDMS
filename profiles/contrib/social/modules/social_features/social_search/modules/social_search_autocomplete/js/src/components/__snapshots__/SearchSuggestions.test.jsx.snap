// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSuggestions renders a list of suggestions 1`] = `
Array [
  <div
    className="search-suggestions"
  >
    <a
      className="search-suggestion"
      href="http://example.com/node/1"
      tabIndex="0"
    >
      <div
        className="search-suggestion__type"
      >
        Topic
      </div>
      <div
        className="search-suggestion__body"
      >
        <div
          className="search-suggestion__label"
        >
          
          <b>
            Open
          </b>
           Social is great for customisable communities
        </div>
      </div>
    </a>
    <a
      className="search-suggestion"
      href="http://example.com/node/2"
      tabIndex="0"
    >
      <div
        className="search-suggestion__type"
      >
        Event
      </div>
      <div
        className="search-suggestion__body"
      >
        <div
          className="search-suggestion__label"
        >
          
          <b>
            Open
          </b>
           Social launching party
        </div>
        <div
          className="search-suggestion__tags"
        >
          <span
            className="search-suggestion__tag badge badge-default badge--pill"
          >
            You are enrolled to this event
          </span>
        </div>
      </div>
    </a>
    <a
      className="search-suggestion"
      href="http://example.com/node/3"
      tabIndex="0"
    >
      <div
        className="search-suggestion__type"
      >
        Event
      </div>
      <div
        className="search-suggestion__body"
      >
        <div
          className="search-suggestion__label"
        >
          GoalGorilla anniversary
        </div>
      </div>
    </a>
  </div>,
  <div
    className="search-suggestions__all"
  >
    <a
      className="btn btn-default btn-raised"
      href="http://example.com/search/base/open"
    >
      See all results
    </a>
  </div>,
]
`;
